frontend/lib/uast/post-parse-checks.cpp

```diff
@@ -1972,6 +1970,56 @@ void Visitor::checkIterNames(const Function* node) {
   return;
 }
 
+struct CountReturnsYields {
+  const Function* symbol = nullptr;
+  int nReturnSomething = 0;
+  int nReturnEmpty = 0;
+  const Return* firstReturnSomething = nullptr;
+  const Return* firstReturnEmpty = nullptr;
```

Comment:
This is called `CountReturnsYields` but only seems to count returns. Moreover, it's certainly not equivalent to the yield / return checks that were removed from the production AST checking code. Did you mean to include that in this PR as well? Alternatively, is there a reason the checks are no longer needed?

Reply:
I can rename it to `CountReturns`.

Those checks already existed in the form of checks within `checkParentsForControlFlow`. In fact I had at first assumed that `checkParentsForControlFlow` covered all of them, but the `Not all returns in this function return a value` error was not (as revealed by testing).

Are the others covered? I can check:
 * `return statement is not in a function or iterator` -- covered by `'return' is not allowed outside of a procedure or an iterator`
 * `returning a value in an iterator` -- covered by `return' statements with values are not allowed in iterators`
 * `yield statement is outside an iterator` -- covered by `'yield' is not allowed outside of an iterator`
 * `yield statement is in a non-iterator function` -- covered by `'yield' is not allowed outside of an iterator`